File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int l = 0, r = height.length - 1, best = -1;
        while(l < r){
            int length = r-l;
            int lowestHeight = Math.min(height[l], height[r]);
            int area = length * lowestHeight;
            if(best < (area)){
                best = area;
            }
            if(height[l] < height[r]){l++;}
            else{r--;}
        }
        return best;
    }
}

File: 11-container-with-most-water/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let left = 0; let right = height.length-1;
    let maxWater = 0;
    while(left < right){
        let minHeight = Math.min(height[left], height[right]);
        let length = right-left;
        let currAmountOfWater = minHeight * length;
        maxWater = currAmountOfWater > maxWater ? currAmountOfWater : maxWater;
        if(height[right] < height[left]){
            right--;
        }
        else{left++;}
    }
    return maxWater;
};